src/Nethermind/Nethermind.Core/Collections/ValueHashToHash256Comparer.cs

```diff
@@ -0,0 +1,25 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using Nethermind.Core.Crypto;
+
+namespace Nethermind.Core.Collections;
+
+public class ValueHashToHash256Comparer : IEqualityComparer<Hash256AsKey>, IAlternateEqualityComparer<ValueHash256, Hash256AsKey>
```

Comment:
There is already `Hash256AsKeyComparer` in `Nethermind.Core.Crypto` (we can move it here)

Reply:
Used previous